### PR TITLE
Oauth2

### PR DESCRIPTION
**Pull request by aravind289**
Added:
1. Django-all-auth: python package used for authentication and registration using 3rd party such as github,facebook google.In this added only gmail.
2. django-rest-auth: package that is used to provide endpoint-apis for 3rd party authorisation
Modified:added the configuration for django-all-auth and added the endpoint for  it

### CODE REVIEW

File: typebackend/socialconnectviews.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
+# from allauth.socialaccount.providers.facebook.views import FacebookOAuth2Adapter
+# from allauth.socialaccount.providers.github.views import GitHubOAuth2Adapter
+# from allauth.socialaccount.providers.twitter.views import TwitterOAuthAdapter
+# from allauth.socialaccount.providers.oauth2.client import OAuth2Client
+# from rest_auth.registration.views import SocialConnectView
+# from rest_auth.social_serializers import TwitterConnectSerializer
+
+# class FacebookConnect(SocialConnectView):
+#     adapter_class = FacebookOAuth2Adapter
+
```
Comment (digi0ps) [inline typebackend/socialconnectviews.py:10]: Is this file needed?

File: typerforslack/settings.py [hunk 2]
```diff
@@ -27,7 +27,8 @@
 # Database
 # https://docs.djangoproject.com/en/3.0/ref/settings/#databases
 
-LOCAL_URl = 'postgres://sanjay:typeit@localhost/typeforslack'
+LOCAL_URl = 'postgres://aravind:password@localhost/typeforslack'
```
Comment (digi0ps) [inline typerforslack/settings.py:30]: Let's change this to a common username like `admin`

File: typebackend/socialconnectviews.py (hunk 1 above)
Comment (aravind289) [reply typebackend/socialconnectviews.py:10]: No.i mean if we want we can use it for the user to connect using other social media login too

File: typerforslack/settings.py (hunk 2 above)
Comment (aravind289) [reply typerforslack/settings.py:30]: oh yes..need to be a standard one
